### PR TITLE
Npm package 1.2 (adding npm package support for microsoft team client sdk)

### PR DESCRIPTION
**Pull request by sachingupta**
These changes enable us to create npm package Bhargav Krishna for helping with these changes

### CODE REVIEW

File: src/MicrosoftTeams.ts [hunk 1]
```diff
@@ -850,13 +850,13 @@ namespace microsoftTeams {
...
-          width +
-          ", height=" +
-          height
+        top +
+        ", left=" +
+        left +
+        ", width=" +
+        width +
+        ", height=" +
+        height
```
Comment (WrathOfZombies) [inline src/MicrosoftTeams.ts]: Undo this change. As its not necessary for this PR

**Comment by WrathOfZombies** [comment]
Please seek approval from @ydogandjiev or @jayongg before merging.

File: .gitignore [hunk 2]
```diff
@@ -1,6 +1,8 @@
 # Dependency directory
 # https://www.npmjs.org/doc/misc/npm-faq.html#should-i-check-my-node_modules-folder-into-git
 node_modules/
+npm-debug
+npm-error
 
 # Typings
 typings/
```
Comment (ydogandjiev) [inline .gitignore]: This can probably be removed

File: gulpfile.js [hunk 3]
```diff
@@ -19,31 +21,32 @@ var AuthenticationContext = require("adal-node").AuthenticationContext;
...
+}
+`;
 
 /// global options
 var options = {
   connectionString: ""
 };
 
-gulp.task("typings", function() {
+gulp.task("typings", function () {
```
Comment (ydogandjiev) [inline gulpfile.js]: Get rid of this if it's not being used

File: gulpfile.js [hunk 4]
```diff
@@ -4,9 +4,11 @@ var del = require("del");
...
 var gulp = require("gulp");
 var gutil = require("gulp-util");
+var umd = require('gulp-umd');
 var karma = require("karma").Server;
 var merge = require("merge2");
 var tslint = require("gulp-tslint");
+var header = require('gulp-header');
 var typescript = require("gulp-typescript");
 var rename = require("gulp-rename");
 var typings = require("gulp-typings");
```
Comment (ydogandjiev) [inline gulpfile.js]: Probably not needed
